factory-method/src/main/java/com/snr/factory/method/VinylCreator.java

```diff
@@ -0,0 +1,13 @@
+package com.snr.factory.method;
+
+public class VinylCreator {
+    public Vinyl create(String type){
+        if(type.equals("LP")){
+            return new LP();
+        } else if(type.equals("EP")){
+            return new EP();
+        } else {
+            return null;
```

Comment:
hier besser eine exception werfen (ansonsten wird der aufrufer meinen er habe ein sauberes objekt zurückerhalten) und so in einen nullpointer laufen:)